pkg/capabilities/oci/manifest/types.go

```diff
@@ -0,0 +1,12 @@
+package manifest
+
+import specs "github.com/opencontainers/image-spec/specs-go/v1"
+
+// We don't need to expose that to consumers of the library
+// This is a glorified wrapper needed to unmarshal a string
+// inside of TinyGo. As of release 0.29.0, unmarshal a simple
+// string causes a runtime panic
+type OciImageManifestResponse interface {
```

Comment:
Can you describe what is going on with TinyGo? BTW, 0.30 is out since some time, have you tried with it?

Aside from that, can't we make this a type, instead of a struct?

@fabriziosestito: do you know any idiomatic way to achieve something like a Rust enum with Go? I fear this is not doable.

The problem we have here is that a remote manifest could be either a `spec.Manifest` or a `spec.Index`

Reply:
> Can you describe what is going on with TinyGo? BTW, 0.30 is out since some time, have you tried with it?

The comment is wrong. It just a copy-paste issue. I've just removed.

Reply:
> Aside from that, can't we make this a type, instead of a struct?
> 
> @fabriziosestito: do you know any idiomatic way to achieve something like a Rust enum with Go? I fear this is not doable.
> 
> The problem we have here is that a remote manifest could be either a `spec.Manifest` or a `spec.Index`

Another option is something like this:
```go
package main

import (
    "encoding/json"
    "fmt"
)

type StructA struct {
    FieldA1 string `json:"field_a1"`
    FieldA2 int    `json:"field_a2"`
}

type StructB struct {
    FieldB1 string  `json:"field_b1"`
    FieldB2 float64 `json:"field_b2"`
}

type UnionStruct struct {
    *StructA
    *StructB
}

func parseData(data []byte) (*UnionStruct, error) {
    var u UnionStruct
    if err := json.Unmarshal(data, &u); err != nil {
        return nil, err
    }
    return &u, nil
}

func main() {
    jsonData := []byte(`{"field_a1": "value1", "field_a2": 123, "field_b2": 456}`)

    parsedData, err := parseData(jsonData)
    if err != nil {
	fmt.Println("error", err)
        return
    }

    if parsedData.StructA != nil {
        fmt.Println("Parsed as StructA:", *parsedData.StructA)
    }
    if parsedData.StructB != nil {
        fmt.Println("Parsed as StructB:", *parsedData.StructB)
    }
}
```

Reply:
> @fabriziosestito: do you know any idiomatic way to achieve something like a Rust enum with Go? I fear this is not doable.

There is no way to achieve this without abusing golang interfaces (https://github.com/BurntSushi/go-sumtype). I've seen this used in the while by code generators (e.g. google protobuf types) but I wouldn't go this way.

I think having two methods is the most idiomatic thing we can do in this case:
 `GetOCIIndexManifest` `GetOCIImageManifest`

Reply:
@fabriziosestito  We cannot have two methods, the problem is the remote OCI registry has an API that returns two types of objects, depending on what the user pushed.

Reply:
Ok, we can't have anything better than that with Go. Let's leave the current implementation.

@jvanz can you make `OciImageManifestResponse` a type instead of an interface?